packages/api/src/apps/datastore_api/app.ts

```diff
@@ -154,27 +156,6 @@ export default async (env: ProcessEnv) => {
     globalStats: Services.stats.Global({ debug }, appState),
   };
 
-  const initBlock = await provider.getBlock("latest");
-
-  // warm caches
-  await services.registry(appState.lastBlockUpdate, initBlock.number);
```

Comment:
this might be tricky but these state updates need to run in a certain order, and the loops introduce some race conditions.  would feel good if you could run a test against empty DB and make sure it eventually gets into a good state. we can also test this on dev and do queries against it